src/ripple/app/tx/impl/Payment.cpp

```diff
@@ -477,7 +477,7 @@ Payment::doApply()
         //  3. If the destination's XRP balance is
         //    a. less than or equal to the base reserve and
         //    b. the deposit amount is less than or equal to the base reserve,
-        // then we allow the deposit.
+        // then we allow the deposit unless the destination is AMM account.
```

Comment:
I think it would be clearer and simpler if the AMM check is separate from this block entirely. Another consideration is that non-mainnet networks might not have `featureDepositAuth` enabled, which is needed for `reqDepositAuth` to be `true`.

In other words, undo the two changes in this commit in this file, and add a separate AMM check.
```
// The source account does have enough money.  Make sure the
// source account has authority to deposit to the destination.

// AMMs can never receive an XRP payment. Must use AMMDeposit transaction instead.
if (sleDst->getFlags() & lsfAMM)
    return tecNO_PERMISSION;

if (reqDepositAuth)
{
    // If depositPreauth is enabled, then an account that requires
[...]
```